tf/scripts/backend_user_data.sh

```
@@ -1,20 +1,218 @@
+log "Building Docker image..."
 cd /home/ec2-user/backend
-sudo docker build -t lti-backend .
+docker build -t lti-backend .
+
+# Run Docker container
+log "Starting Docker container..."
+docker run -d -p 8080:8080 --name lti-backend-container lti-backend
+
+# Install Datadog Agent
+log "Installing Datadog Agent..."
+export DD_API_KEY="${datadog_api_key}"
```

Shellcheck (0.11.0)

[warning] datadog_api_key is referenced but not assigned.

(SC2154)